mutable_tree.go

```diff
@@ -117,33 +117,41 @@ func (tree *MutableTree) String() (string, error) {
 
 // Set/Remove will orphan at most tree.Height nodes,
 // balancing the tree after a Set/Remove will orphan at most 3 nodes.
-func (tree *MutableTree) prepareOrphansSlice() []*Node {
-	return make([]*Node, 0, tree.Height()+3)
+func (tree *MutableTree) prepareOrphansSlice() []OrphanedNode {
+	return make([]OrphanedNode, 0, tree.Height()+3)
+}
+
+
+
+type OrphanedNode interface {
```

Comment:
If the node's interface/struct does not have a file dedicated to it, let's have it in one place - `node.go`